snippets/product-thumbnail.liquid

```diff
@@ -20,56 +21,43 @@
     %}
 {% endcomment %}
 
+{%- liquid
+  unless lazy_load == false
+    assign lazy = 'lazy'
+  endunless
+
+  assign desktop_columns = 1
+  if desktop_layout == 'columns'
+    assign desktop_columns = 2
+  endif
+-%}
+
+{%- capture sizes -%}
+  (min-width: {{ settings.page_width }}px) {{ settings.page_width | minus: 100 | times: media_width | divided_by: desktop_columns | round }}px, (min-width: 990px) calc({{ media_width | times: 100 | divided_by: desktop_columns }}vw - 10rem), (min-width: 750px) calc((100vw - 11.5rem) / 2), calc(100vw - 4rem)
```

Comment:
Oh I just realized the `destkop_columns` is not accounting for when there is only one image 🤔 

![](https://screenshot.click/25-15-37i4l-uac9s.png)

Reply:
So here when comparing the image used, when 2 columns it's `990` and when the media layout is set to stacked it's using a `1426` version of the image.

Reply:
https://github.com/Shopify/dawn/pull/1766/commits/28109531769855c08a9e767ee2dbfe6d8f9b3e1a

I passed the `section.settings.hide_variants` setting to product thumbnail so that for products with only 1 image, and using 2 columns + "Hide other variants’ media after selecting a variant", the image size will be the same as using the "stacked" layout. Let me know if there's a simpler way of doing this.